R/transition_risk_score.R

```diff
@@ -0,0 +1,126 @@
+    create_benchmarks_averages() |>
+    select(-c("transition_risk_score", "reduction_targets", "profile_ranking")) |>
+    distinct()
+
+  nest_levels(trs_product, trs_company)
+}
+
+create_tr_benchmarks_tr_score <- function(data) {
+  mutate(data,
+    transition_risk_score = ifelse(is.na(.data$profile_ranking) | is.na(.data$reduction_targets), NA,
+      (.data$profile_ranking + .data$reduction_targets) / 2
+    ),
+    benchmark_tr_score = ifelse(is.na(.data$scenario_year) | is.na(.data$benchmark), NA,
+      paste(.data$scenario_year, .data$benchmark, sep = "_")
+    )
+  )
+}
+
+create_benchmarks_averages <- function(data) {
+  mutate(data,
+    transition_risk_score_avg = mean(.data$transition_risk_score, na.rm = TRUE),
+    reduction_targets_avg = mean(.data$reduction_targets, na.rm = TRUE),
+    profile_ranking_avg = mean(.data$profile_ranking, na.rm = TRUE),
+    .by = c("companies_id", "benchmark_tr_score")
+  )
+}
+
+prepare_trs_emissions <- function(data) {
+  select(data, c(
+    "companies_id", "benchmark", "profile_ranking", "ep_product",
+    "activity_uuid_product_uuid"
+  ))
+}
+
+prepare_trs_sector <- function(data) {
+  data |>
+    select(c(
+      "companies_id", "scenario", "year", "reduction_targets", "ep_product",
+      "activity_uuid_product_uuid"
+    )) |>
+    mutate(scenario_year = paste(.data$scenario, .data$year, sep = "_")) |>
+    select(-c("scenario", "year"))
+}
+
+full_join_emmissions_sector <- function(emissions, sector) {
+  full_join(emissions, sector,
+    by = c("companies_id", "ep_product", "activity_uuid_product_uuid"),
+    relationship = "many-to-many"
+  )
+}
+
+get_rows_union_for_common_cols <- function(emissions_at_product_level, sector_at_product_level) {
+  emission_common_columns <- emissions_at_product_level |>
+    select(common_columns_emissions_sector()) |>
+    distinct()
+
+  sector_common_columns <- sector_at_product_level |>
+    select(common_columns_emissions_sector()) |>
+    distinct()
+
+  distinct(bind_rows(emission_common_columns, sector_common_columns))
+}
+
+relocate_trs_product_columns <- function() {
+  c(
+    "companies_id", "company_name", "country", "transition_risk_score", "benchmark_tr_score",
+    "profile_ranking", "reduction_targets"
+  )
+}
+
+trs_company_columns <- function() {
+  c(
+    "companies_id", "company_name", "country", "company_city", "transition_risk_score",
+    "benchmark_tr_score", "profile_ranking", "reduction_targets", "postcode", "address",
+    "main_activity"
+  )
+}
+
+common_columns_emissions_sector <- function() {
+  c(
+    "companies_id", "company_name", "country", "ep_product", "matched_activity_name",
+    "matched_reference_product", "unit", "multi_match", "matching_certainty",
+    "matching_certainty_company_average", "company_city", "postcode", "address",
+    "main_activity", "activity_uuid_product_uuid", "tilt_sector", "tilt_subsector",
+    "isic_4digit", "isic_4digit_name", "ei_geography"
+  )
```

Comment:
Maybe run a styler so that each column takes it's own row. This makes the future git diffs are easier to see.

Reply:
Thanks. If the automatic styler doesn't touch it then great. 

It's a soft "nice to have" but I tend to prefer to stick as much as possible to a width of 80 characters. That's keeps the code complient with the tidyverse style guide and with linters (e.g. `goodpractice::goodpractice()`).

One way to easily reformat the code is with Shift + Ctrl + A.

You can find it in the commands pallete (Shift + Ctrl + P) as "Reformat Current Selection".

![image](https://github.com/2DegreesInvesting/tiltIndicatorAfter/assets/5856545/793bb7bd-fb65-43a3-ab0a-e07b90718ae7)

Reply:
Thanks for this insight! Very helpful :)